src/pymoca/parser.py

```diff
@@ -1037,8 +1037,11 @@ def parse(
             (datetime.utcnow() - timedelta(days=cache_expiration_days)).timestamp() * 1e6
         )
```

Comment:
This is a side issue, but I just noticed:

1. The `cutoff_time` calc should use `_microseconds_since_epoch()` instead of copy/paste code.
2. `datetime.utcnow()` used here and two other places is deprecated as of Python 3.12. They suggest using `datetime.now(datetime.timezone.utc)` instead.

Reply:
I added a bunch of commits on top to address both these